pkg/ddevapp/config.go

```diff
@@ -1334,7 +1315,33 @@ fi`, app.Database.Version, app.GetStartScriptTimeout(), psqlVersion) + "\n\n"
 	// Assets in the web-build directory copied to .webimageBuild so .webimageBuild can be "context"
 	// This actually copies the Dockerfile, but it is then immediately overwritten by WriteImageDockerfile()
 	if userDockerfilePath != "" {
-		err = copy2.Copy(userDockerfilePath, filepath.Dir(fullpath))
+		err = copy2.Copy(userDockerfilePath, filepath.Dir(fullpath), copy2.Options{Skip: func(_ os.FileInfo, src, _ string) (bool, error) {
```

Comment:
Pretty fancy! Please add a comment above this line saying what it's doing and why, thanks.

Would be be better to turn this into an actual function instead of a closure?

Reply:
I cannot get rid of closure completely, because I need `userDockerfilePath` inside that closure to detect top-level directory (so I can remove `*.example` files correctly, without touching subdirs).

Is it okay to use something like this?:

```go
err = copy2.Copy(userDockerfilePath, filepath.Dir(fullpath), copy2.Options{
	Skip: func(_ os.FileInfo, src, _ string) (bool, error) {
		return shouldSkipDockerfileOnCopy(src, userDockerfilePath)
	},
})
```

Reply:
Yes, works for me. Just want future me to be able to easily understand it.  I would have accepted as it was, just looking for clearer presentation.